yoti_test.go

```diff
@@ -72,10 +72,12 @@ func TestYotiClient_HttpFailure_ReturnsFailure(t *testing.T) {
 		},
 	}
 
-	_, _, errorStrings := client.getActivityDetails(encryptedToken)
+	_, _, err := client.getActivityDetails(encryptedToken)
 
-	assert.Assert(t, len(errorStrings) > 0)
-	assert.Check(t, strings.HasPrefix(errorStrings[0], "Unknown HTTP Error"))
+	assert.Check(t, err != nil)
+	assert.Check(t, strings.HasPrefix(err.Error(), "Unknown HTTP Error"))
+	_, temporary := err.(TemporaryError)
+	assert.Check(t, temporary)
```

Comment:
The tests would be more flexible here if you check for `interface{ Temporary() bool}` you would also wan't to check the return of Temporary.